File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.java
class Solution {
    public int garbageCollection(String[] garbage, int[] travel) {
        int ans=0;
        int f=0;
        for(int i=0;i<garbage.length;i++){
            if(garbage[i].contains("P")){
                f=i;
                for(int j=0;j<garbage[i].length();j++){
                    if(garbage[i].charAt(j)=='P'){
                        ans++;
                    }
                }
            }
        }
        for(int i=0;i<f;i++){
            ans=ans+travel[i];
        }
        f=0;
        for(int i=0;i<garbage.length;i++){
            if(garbage[i].contains("G")){
                f=i;
                for(int j=0;j<garbage[i].length();j++){
                    if(garbage[i].charAt(j)=='G'){
                        ans++;
                    }
                }
            }
        }
        for(int i=0;i<f;i++){
            ans=ans+travel[i];
        }
        f=0;
        for(int i=0;i<garbage.length;i++){
            if(garbage[i].contains("M")){
                f=i;
                for(int j=0;j<garbage[i].length();j++){
                    if(garbage[i].charAt(j)=='M'){
                        ans++;
                    }
                }
            }
        }
        for(int i=0;i<f;i++){
            ans=ans+travel[i];
        }
        return ans;
    }
}

File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.cpp
class Solution {
public:
    int garbageCollection(vector<string>& garbage, vector<int>& travel) {
        int count =0;
          int index=0;

        for(int i =0; i<garbage.size() ; i++){
            for(int j =0; j<garbage[i].size() ; j++){
                char ch = garbage[i][j];

                if(ch=='M'){
                    count+=1;
                    index=i;
                }
            }


        }
        for(int i=0;i<index;i++){
            count=count+travel[i];
        }
        index=0;
        for(int i =0; i<garbage.size() ; i++){
            for(int j =0; j<garbage[i].size() ; j++){
                char ch = garbage[i][j];

                if(ch=='P'){
                    count+=1;
                    index=i;

                }
            }
        }
        for(int i=0;i<index;i++){
            count=count+travel[i];
        }
        index=0;
        for(int i =0; i<garbage.size() ; i++){
            for(int j =0; j<garbage[i].size() ; j++){
                char ch = garbage[i][j];

                if(ch=='G'){
                    count+=1;
                    index=i;

                }
            }

        }
        for(int i=0;i<index;i++){
            count=count+travel[i];
        }
        return count;
    }
};